src/apikey/middlewares/auth.js

```diff
@@ -0,0 +1,25 @@
+const config = require('config');
+
+/**
+ * check the user's api key
+ * @param {Object} req - HTTP request
+ * @param {Object} res - HTTP response
+ * @param {function} next - do the following
+ * @returns {Object|function} res or next
+ */
+const checkAuth = async (req, res, next) => {
+  // TODO check in query
+  const apikey = req.get('redis-password');
+
+  if (!apikey) {
+    return res.status(401).json({ message: 'Not authorized' });
+  }
+
+  if (apikey !== config.get('redis.password')) {
+    return res.status(401).json({ message: 'Not authorized' });
+  }
```

Comment:
```suggestion
  if (apikey !== config.get('redis.password')) {
    return res.status(401).json({ message: 'Not authorized' });
  }
```
If `apikey` is not set, it won't match the password anyway. However you can also write `if (!apikey || apikey !== config.get('redis.password'))` to cover the case where `redis.password` is not set, without duplicating code.